vm/src/stdlib/signal.rs

```diff
@@ -279,4 +279,21 @@ pub(crate) mod _signal {
             Err(vm.new_value_error("signal number out of range".to_owned()))
         }
     }
+
+    /// Similar to `PyErr_SetInterruptEx` in CPython
+    ///
+    /// Missing signal handler for the given signal number is silently ignored.
+    #[allow(dead_code)]
+    pub fn set_interrupt_ex(signum: i32, vm: &VirtualMachine) -> PyResult<()> {
+        assert_in_range(signum, vm)?;
+
+        match signum as usize {
+            SIG_DFL | SIG_IGN => Ok(()),
+            _ => {
+                // interrupt the main thread with given signal number
+                run_signal(signum);
+                Ok(())
+            }
+        }
+    }
```

Comment:
we put internal signal tools to `vm/src/signal.rs` and python functions to `vm/src/stdlib/signal.rs`. This is better to be pu in `vm.src/signal.rs`